analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/TestHelper.cs

```diff
@@ -218,16 +218,14 @@ public static AnalyzerOptions CreateOptions(string relativePath, SourceText text
         public static string ToUnixLineEndings(this string value) =>
             value.Replace(Constants.WindowsLineEnding, Constants.UnixLineEnding);
 
-        public static string CreateFilesToAnalyze(string filesToAnalyzeDirectory, params string[] filesToAnalyze)
-        {
-            var filestoAnalyzePath = Path.Combine(filesToAnalyzeDirectory, "FilesToAnalyze.txt");
-            File.WriteAllLines(filestoAnalyzePath, filesToAnalyze);
-            return filestoAnalyzePath;
-        }
-
         public static string TestPath(TestContext context, string fileName)
         {
-            var path = Path.Combine(context.TestDir, context.FullyQualifiedTestClassName.Replace("SonarAnalyzer.UnitTest.", null), context.TestName, "TestCases", fileName);
+            var path = Path.Combine(context.TestDir, context.FullyQualifiedTestClassName.Replace("SonarAnalyzer.UnitTest.", null), context.TestName, fileName);
+            if (path.Length > 250)  // 260 can throw PathTooLongException
+            {
+                var directory = Path.Combine(context.TestDir, context.FullyQualifiedTestClassName.Replace("SonarAnalyzer.UnitTest.", null));
+                path = Path.Combine(directory, $"TooLongTestName.{Directory.GetDirectories(directory).Length}", fileName);
+            }
             Directory.CreateDirectory(Path.GetDirectoryName(path));
             return path;
         }
```

Comment:
I would go for something similar with:
```suggestion
            var parent = Path.Combine(context.TestDir, context.FullyQualifiedTestClassName.Replace("SonarAnalyzer.UnitTest.", null));
            var folderName = parent.Length + context.TestName.Length + fileName.Length > 250 // 260 can throw PathTooLongException
                                 ? $"TooLongTestName.{Directory.GetDirectories(parent).Length}"
                                 : context.TestName;
            var path = Path.Combine(parent, folderName, fileName);
            Directory.CreateDirectory(Path.GetDirectoryName(path));
            return path;
```
We avoid computing the first part twice and I think it make the intention clearer.